src/telemetry.rs

```diff
@@ -44,32 +71,33 @@ impl Telemetry {
     /// We are using `impl Subscriber` as return type to avoid having to spell out the actual
     /// type of the returned subscriber, which is indeed quite complex.
     pub fn init(
-        name: String,
+        resource: Resource,
         tracing_settings: TracingSettings,
     ) -> anyhow::Result<(Self, impl Subscriber + Sync + Send)> {
         global::set_text_map_propagator(TraceContextPropagator::default());
 
+        let name = resource.get(semcov::resource::SERVICE_NAME);
```

Comment:
Do you use `name` now? Maybe remove it.

Reply:
It used by formatter a few lines below. It should be created here, because `trace!` macro consumes `resource`;